### PR TITLE
Fix our rubocop setup

### PR DESCRIPTION
**Pull request by eduardoj**
Co-authored-by: Björn Geuken <bgeuken@suse.de>

### CODE REVIEW

**Comment by eduardoj** [comment]
[P13: Fix Rubocop setup](https://trello.com/c/ooegd1Ej/784-p13-fix-rubocop-setup)

File: contrib/.rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+inherit_from: .rubocop_todo.yml
```
Comment (Ana06) [inline contrib/.rubocop.yml]: why do we have a `.rubocop.yml` file in `contrib/` and another one in `dist/`? You can just have one, you only need to split the rails directory... :confused:

File: .rubocop.yml [hunk 2]
```diff
@@ -1,25 +1,3 @@
-require: rubocop-rspec
```
Comment (Ana06) [inline .rubocop.yml]: the rules in `.rubocop.yml` are not used... you are now enabling for example `RedundantReturn` which we agreed to disable

File: src/api/lib/tasks/dev.rake [hunk 3]
```diff
@@ -72,15 +72,31 @@ namespace :dev do
...
     sh 'jshint ./app/assets/javascripts/'
     Rake::Task['db:structure:verify'].invoke
     Rake::Task['db:structure:verify_no_bigint'].invoke
   end
   namespace :lint do
-    desc 'Run the ruby linter'
-    task :ruby do
-      sh 'rubocop -D -F -S --fail-level convention ../..'
+    namespace :rubocop do
+      task all: [:contrib, :dist, :rails] do
```
Comment (Ana06) [inline src/api/lib/tasks/dev.rake]: this is not what we want. We want to run it twice: one for the rails directory and one for the rest. Only the Rails directory is special, splitting the rest is extra work without no reason. It is what is in the card as well.

File: src/api/.rubocop_todo.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config`
-# on 2018-06-20 13:49:50 +0000 using RuboCop version 0.57.2.
+# on 2018-06-20 14:26:57 +0000 using RuboCop version 0.57.2.
```
Comment (Ana06) [inline src/api/.rubocop_todo.yml]: why is this in the [_Add rake task for updating rubocop todo files_ commit](https://github.com/openSUSE/open-build-service/pull/5173/commits/e3673bc54c91a7b3d483cb9d4c00b5fe0b5ed2cf)? :confused:

File: src/api/lib/tasks/dev.rake [hunk 5]
```diff
@@ -72,16 +72,56 @@ namespace :dev do
...
     Rake::Task['db:structure:verify'].invoke
     Rake::Task['db:structure:verify_no_bigint'].invoke
   end
+
   namespace :lint do
-    desc 'Run the ruby linter'
-    task :ruby do
-      sh 'rubocop -D -F -S --fail-level convention ../..'
+    namespace :rubocop do
+      task all: [:contrib, :dist, :rails] do
```
Comment (ChrisBr) [inline src/api/lib/tasks/dev.rake]: This is a nice idea :+1:

**Comment by bgeuken** [comment]
> And the title: Refactor rubocop setup.. this is not a refactoring... it is fixings it...

That was automatically created based on the branch name. Could it be that you put to much emphasize on the wrong things?:-)

File: src/api/lib/tasks/dev.rake (hunk 3 above)
Comment (bgeuken) [reply src/api/lib/tasks/dev.rake]: Seems we could add the '--ignore-parent-exclusion' parameter to make this work. @eduardoj let's try that tomorrow

**Comment by eduardoj** [comment]
Superseded by #5182

File: src/api/lib/tasks/dev.rake (hunk 3 above)
Comment (Ana06) [reply src/api/lib/tasks/dev.rake]: @bgeuken 

> Seems we could add the '--ignore-parent-exclusion' parameter to make this work. 

exactly... and even without that you could rename the files... :una

**Comment by Ana06** [comment]
@bgeuken 

> That was automatically created based on the branch name. Could it be that you put to much emphasize on the wrong things?:-)

No... I just said what I think is wrong. No emphasize...

**Comment by Ana06** [comment]
@eduardoj 

> Superseded by #5182

why? :unamused:

**Comment by eduardoj** [comment]
@Ana06 :

> why? :unamused:

Used a different approach. Instead of having the configuration spread in each subdirectory now it is only spread in two rubocop configuration files.

If you prefer I can also restore this original branch and put the new three commits here. Not a big deal.

**Pull request closed without merging**